### DISCUSSION TITLE
Support for new Cisco IOL images

### DISCUSSION THREAD

**Discussion by fluzzi** [srl-labs/containerlab#2150] [Ideas]
The new Cisco IOL images that run directly on linux, supports 64 bits OS. I've created a Dockerfile and some scripts that make it work containerized and it works with containerlab.

this images are really great to run IOS as they don't consume as much resources and are not VMs.

https://github.com/fluzzi/iol_docker

while my workaround works, it's not ideal and has some flaws So I would like to request direct support for this.

thanks!

**Comment by fluzzi** [comment]
This is how i make it work on containerlab:

    iol:
      kind: linux
      image: iol-container
      binds:
        - ./nvram:/opt/iol/nvram_00001
      mgmt-ipv4: 10.30.31.45

**Comment by hellt** [comment]
Is it possible to download IOL somewhere? If yes, we can document the whole process in the docs

**Reply by fluzzi** [reply]
usually from Cisco modeling labs 2.7 but you can get it from a [link](https://upw.io/4JY/iosxe-iol-2.7.rar)

**Comment by kaelemc** [comment]
@fluzzi That's awesome!. I was attempting to reverse engineer IOL based on CML and found this, awesome that you have done the work. 

Are you happy for me to work off/modify what you've done so far, modify it and add it to containerlab?

**Reply by fluzzi** [reply]
That would be great! That was what i was aiming for. Let me know if I can help in any way or clarify what I did.

**Reply by kaelemc** [reply]
Hey @fluzzi. Sorry for the delay but I have made a [PR](https://github.com/hellt/vrnetlab/pull/256) under the vrnetlab repo (and a containerlab one is coming soon). 

It turns out a .iourc file wasn't required for me to launch IOL. 

If you're able to test, kindly let me know if this is the same for you?
